sprint-poker-springboot/src/main/java/org/zhxie/sprinpoker/controller/TicketRecordController.java

```diff
@@ -35,35 +38,52 @@
     }
 
     @RequestMapping( method = RequestMethod.POST)
-    public Boolean save(@RequestBody TicketRecord ticketRecord) {
+    public ResponseResult save(@RequestBody TicketRecord ticketRecord) {
+        String roles = (String) request.getAttribute("roles");
+        if (Strings.isNullOrEmpty(roles)) {
+            return new ResponseResult(ResponseResult.NEED_LOGIN, "请先登录");
+        }
         ticketRecord.setId(null);
         ticketRecord.setDate(LocalDate.now());
-        TicketRecord result = ticketRecordService.save(ticketRecord);
-        return result != null;
+        ticketRecordService.save(ticketRecord);
+        return new ResponseResult(ResponseResult.SUCCESS, "保存成功！");
     }
 
     @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
-    public Boolean update(@PathVariable int id, @RequestBody TicketRecord ticketRecord) {
+    public ResponseResult update(@PathVariable int id, @RequestBody TicketRecord ticketRecord) {
+        String roles = (String) request.getAttribute("roles");
+        if (Strings.isNullOrEmpty(roles)) {
+            return new ResponseResult(ResponseResult.NEED_LOGIN, "请先登录");
+        }
+
         Optional<TicketRecord> optional = ticketRecordService.findById(id);
         if (!optional.isPresent()){
-            return false;
+            return new ResponseResult(ResponseResult.FAIL, "记录不存在");
         }
         ticketRecord.setId(id);
         ticketRecord.setDate(optional.get().getDate());
-        TicketRecord result = ticketRecordService.save(ticketRecord);
-        return result != null;
+        ticketRecordService.save(ticketRecord);
+        return new ResponseResult(ResponseResult.SUCCESS, "修改成功！");
     }
 
     @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
-    public Boolean deleteById(@PathVariable int id) {
+    public ResponseResult deleteById(@PathVariable int id) {
+        String roles = (String) request.getAttribute("roles");
+        if (Strings.isNullOrEmpty(roles)) {
```

Comment:
you should not check the role in this way. We should use AOP or check in interceptor so that we don't use to check the permission in every api call and mixed the others logic.